projects/cel-cpp/BUILD

```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+cc_binary(
```

Comment:
Can you write this using https://google.github.io/oss-fuzz/getting-started/new-project-guide/bazel/ ?

@stefanbucur fyi.

Reply:
I just tried in branch celcpp2 and failed with 
`ERROR: /root/.cache/bazel/_bazel_root/9237247786220dac906b18800270f37f/external/rules_fuzzing/fuzzing/init.bzl:17:1: file '@rules_python//python:pip.bzl' does not contain symbol 'pip_install'`

This seems to come from
https://github.com/bazelbuild/rules_fuzzing#configuring-the-workspace

@stefanbucur did I do something wrong ?

Reply:
This is quite surprising, since `pip_install` is there in `@rules_python//python:pip.bzl`: https://github.com/bazelbuild/rules_python/blob/c37ba2215eccab53ae1da5f827a335281f81b9e1/python/pip.bzl#L19

Do you happen to import the `rules_python` package directly in your WORKSPACE? That might interfere with the version pulled by `rules_fuzzing`: https://github.com/bazelbuild/rules_fuzzing/blob/e611b02d7bb4b44a8b073980793eb15751cfaa4b/fuzzing/repositories.bzl#L29-L34

Reply:
By the way, I'd be happy to take a look if you have a branch with the new code.

Reply:
Here is my branch https://github.com/catenacyber/oss-fuzz/tree/celcpp2

Reply:
Thanks - I think I understand now. One of your transitive deps (possibly Antlr, [for example](https://github.com/marcohu/rules_antlr/blob/89a29cca479363a5aee53e203719510bdc6be6ff/antlr/repositories.bzl#L385)), imports a rather old version of `rules_python`, which is incompatible with what `rules_fuzzing` is using.

Upgrading the Antlr Python dependency is probably the best solution in the long run, but for now, I think a quick workaround would be for you to import the latest Python rules under a different name, and map that name when importing the fuzzing rules.

For example, something along the lines of:

```
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "fuzzing_rules_python",
    url = "https://github.com/bazelbuild/rules_python/releases/download/0.1.0/rules_python-0.1.0.tar.gz",
    sha256 = "b6d46438523a3ec0f3cead544190ee13223a52f6a6765a29eae7b7cc24cc83a0",
)

http_archive(
    name = "rules_fuzzing",
    sha256 = "a5734cb42b1b69395c57e0bbd32ade394d5c3d6afbfe782b24816a96da24660d",
    strip_prefix = "rules_fuzzing-0.1.1",
    urls = ["https://github.com/bazelbuild/rules_fuzzing/archive/v0.1.1.zip"],
    repo_mapping = {
        "@rules_python": "@fuzzing_rules_python",
    },
)

load("@rules_fuzzing//fuzzing:repositories.bzl", "rules_fuzzing_dependencies")

rules_fuzzing_dependencies()

load("@rules_fuzzing//fuzzing:init.bzl", "rules_fuzzing_init")

rules_fuzzing_init()
```

Reply:
By the way, on a related note, have you considered including these fuzz tests directly in your project's repository? This way, you can do dependency management in a single place, the OSS-Fuzz integration is also simpler (no need to concatenate project files), you also get build targets to test your fuzz test locally, etc.

Reply:
Thanks Stefan.
But with tour workaround, I get to another error
`ERROR: <builtin>: Use of Starlark transition without whitelist attribute '_whitelist_function_transition'. See Starlark transitions documentation for details and usage: @rules_fuzzing//fuzzing/private:binary.bzl NORMAL`
You can test it on the same branch https://github.com/catenacyber/oss-fuzz/tree/celcpp2

> have you considered including these fuzz tests directly in your project's repository? 

Indeed, but I am not a cel-cpp maintainer.
This is the initial integration to oss-fuzz, so I find it simpler to have only one Pull Request to one repository
The next step should indeed be to integrate the fuzz target in the upstream repository
@kyessenov what is your mind about this ?

Reply:
Ah, that error was due to `cel-cpp` pinning an old Bazel version in their `.bazelversion` file, which does not support the features needed by `cc_fuzz_target`. However, it seems we can override that version and things appear to be working. Here are my changes:

```
$ git diff
diff --git a/projects/cel-cpp/.bazelrc b/projects/cel-cpp/.bazelrc
index bdf56348..e9a5054b 100644
--- a/projects/cel-cpp/.bazelrc
+++ b/projects/cel-cpp/.bazelrc
@@ -22,3 +22,8 @@ build --action_env=CXX=clang++
 build:asan-libfuzzer --@rules_fuzzing//fuzzing:cc_engine=@rules_fuzzing//fuzzing/engines:libfuzzer
 build:asan-libfuzzer --@rules_fuzzing//fuzzing:cc_engine_instrumentation=libfuzzer
 build:asan-libfuzzer --@rules_fuzzing//fuzzing:cc_engine_sanitizer=asan
+
+build:oss-fuzz --@rules_fuzzing//fuzzing:cc_engine=@rules_fuzzing_oss_fuzz//:oss_fuzz_engine
+build:oss-fuzz --@rules_fuzzing//fuzzing:cc_engine_instrumentation=oss-fuzz
+build:oss-fuzz --@rules_fuzzing//fuzzing:cc_engine_sanitizer=none
+
diff --git a/projects/cel-cpp/Dockerfile b/projects/cel-cpp/Dockerfile
index f154881b..6c4b67ee 100644
--- a/projects/cel-cpp/Dockerfile
+++ b/projects/cel-cpp/Dockerfile
@@ -23,4 +23,5 @@ COPY BUILD fuzz*.cc $SRC/cel-cpp/fuzz/
 COPY WORKSPACE .bazelrc $SRC/
 RUN cat WORKSPACE >> $SRC/cel-cpp/WORKSPACE
 RUN cat .bazelrc >> $SRC/cel-cpp/.bazelrc
+RUN echo "4.0.0" > $SRC/cel-cpp/.bazelversion
 WORKDIR $SRC/cel-cpp
```

Then everything else appears to be working:
```sh
$ python infra/helper.py build_image cel-cpp
$ python infra/helper.py build_fuzzers --sanitizer address cel-cpp
$ python infra/helper.py run_fuzzer --corpus-dir=/tmp/CORPUS envoy base64_fuzz_test
```

Reply:
Thanks for your help, it works with your patch

Reply:
@stefanbucur it seems to fail with sanitizer undefined.
Should I disable it ? Or do you have a better idea ?

Reply:
Hmm, that seems indeed like a `rules_fuzzing` bug. I've just filed bazelbuild/rules_fuzzing#129 to track it.

As a workaround for now, could you add back the `if [ "$SANITIZER" = "undefined" ]; then ...; fi` clause that you had in your original script? Just that part should be sufficient.